engine/src/main/java/com/datatorrent/stram/engine/StreamingContainer.java

```
@@ -1418,19 +1419,32 @@ public void run()
               logger.error("Voluntary container termination due to an error in operator {}.", currentdi, error);
               operators = new int[]{currentdi.id};
             }
-            umbilical.reportError(containerId, operators, "Voluntary container termination due to an error. " + ExceptionUtils.getStackTrace(error));
-            System.exit(1);
+            try {
+              umbilical.reportError(containerId, operators, "Voluntary container termination due to an error. " + ExceptionUtils.getStackTrace(error));
+            } catch (Exception e) {
+              logger.debug("Fail to log", e);
+            } finally {
+              System.exit(1);
+            }
           } catch (Exception ex) {
             if (currentdi == null) {
               failedNodes.add(ndi.id);
               logger.error("Operator set {} stopped running due to an exception.", setOperators, ex);
               int[] operators = new int[]{ndi.id};
-              umbilical.reportError(containerId, operators, "Stopped running due to an exception. " + ExceptionUtils.getStackTrace(ex));
+              try {
+                umbilical.reportError(containerId, operators, "Stopped running due to an exception. " + ExceptionUtils.getStackTrace(ex));
+              } catch (Exception e) {
```

Doesn't this change the flow where before exception would propagate and now it is swallowed?

---

The catch is in main(), there is no need to propagate it. Otherwise, it would propagate to the default exception handler that writes the exception to stderr.

---

I must be missing something. This is in activate, which is called from deploy.

---

Oops, looked at the wrong place. It is in the run() of an operator thread created inside activate(). Before exception would become an unhandled exception and it would be handled by the ThreadGroup.uncaughtException() that writes exception stack trace to stderr.